### ISSUE TITLE
pre-code tags cause Zammad to swallow content

### ISSUE THREAD

**Issue by MrGeneration** [zammad/zammad#4039]
### Infos:

* Used Zammad version: stable 5.1
* Installation method (source, package, ..): any
* Operating system: any
* Database + version: any
* Elasticsearch version: any
* Browser + version: any
* The used mail client for verification was Roundcube v1.5.2 - Thunderbird was said to be "working". It is unknown how below code is being pasted. (This may technically play it's role)
* This issue was originally found by "uli-heller" from our community: https://community.zammad.org/t/incoming-emails-some-parts-are-deleted/8246

### Expected behavior:

Zammad does not swallow text parts.

### Actual behavior:

Zammad partly removes text passages in specific situations.
In this sample html sanitizing seems to find its victim in `<pre><code></code></pre>` constructs:

![image](https://user-images.githubusercontent.com/6549061/160705511-2178a9a0-7682-4990-97ba-95f7132c608b.png)

The content *is not* lost in raw articles but ticket articles only.

Here's a affected mail your can import for verification your self.
The constructed version of me does not contain some parameters - they didn't seem to matter.
For completness: 
```
<pre style=3D"font-size: 1em; font-family: Consolas, Inconsolata, Courier, =
monospace; line-height: 1.2em; margin: 1.2em 0px;"><code style=3D"font-size=
: 0.85em; font-family: Consolas, Inconsolata, Courier, monospace; margin: 0=
px 0.15em; padding: 0.5em 0.7em; white-space: pre; border: 1px solid #ccccc=
c; background-color: #f8f8f8; border-radius: 3px; display: block !important=
; overflow: auto;">
```

```
MIME-Version: 1.0
Content-Type: multipart/alternative;
 boundary="=_e2d4c04609c2d7bd8a014063cc6120fa"
Date: Tue, 29 Mar 2022 22:44:52 +0200
From: Community Verify <community-verify@zammad-hosting.com>
To: jdoe@exmaple.com
Subject: Test
User-Agent: Roundcube Webmail (THA)
Message-ID: <98a70575b3edfb20d543d57c7bcb982d@zammad-hosting.com>
X-Sender: community-verify@zammad-hosting.com
Organization: Zammad GmbH

--=_e2d4c04609c2d7bd8a014063cc6120fa
Content-Transfer-Encoding: 8bit
Content-Type: text/plain; charset=UTF-8;
 format=flowed



Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod 
tempor incididunt ut labore et dolore magna aliqua. Integer malesuada 
nunc vel risus commodo viverra. In aliquam sem fringilla ut. Enim sed 
faucibus turpis in eu mi bibendum neque egestas. Quis auctor elit sed 
vulputate mi sit amet mauris. Duis tristique sollicitudin nibh sit amet 
commodo nulla facilisi nullam. Egestas sed sed risus pretium quam. 
Maecenas sed enim ut sem. Id leo in vitae turpis massa sed elementum 
tempus. Tincidunt lobortis feugiat vivamus at. Commodo odio aenean sed 
adipiscing. Massa id neque aliquam vestibulum morbi blandit. Mauris 
cursus mattis molestie a. Morbi non arcu risus quis varius quam quisque 
id. Volutpat consequat mauris nunc congue nisi vitae suscipit. Aliquet 
eget sit amet tellus cras adipiscing enim.

Viverra suspendisse potenti nullam ac tortor vitae. Et pharetra pharetra 
massa massa. Amet nisl suscipit adipiscing bibendum est ultricies 
integer quis auctor. Vestibulum morbi blandit cursus risus at ultrices. 
Malesuada nunc vel risus commodo viverra maecenas. Dictum at tempor 
commodo ullamcorper. Purus gravida quis blandit turpis cursus in hac 
habitasse platea. Felis bibendum ut tristique et egestas. Augue lacus 
viverra vitae congue eu consequat ac felis. Nunc eget lorem dolor sed 
viverra ipsum nunc aliquet. Ullamcorper velit sed ullamcorper morbi 
tincidunt ornare massa eget. Est ultricies integer quis auctor elit sed.

apt-get update
Get:1 http://security.ubuntu.com/ubuntu focal-security InRelease [114 
kB]
Hit:2 http://de.archive.ubuntu.com/ubuntu focal InRelease
Hit:3 http://de.archive.ubuntu.com/ubuntu focal-updates InRelease
Get:4 http://10.10.21.205:3207/dprepo/ubuntu experimental/20.04_x86_64/ 
InRelease [3820 B]
Hit:5 http://de.archive.ubuntu.com/ubuntu focal-backports InRelease
Get:6 http://10.10.21.205:3207/dprepo/ubuntu 20.04_x86_64/ InRelease 
[3781 B]
Get:7 http://10.10.21.205:3207/dprepo/ubuntu experimental/20.04_x86_64/ 
Sources [2710 B]
Get:8 http://10.10.21.205:3207/dprepo/ubuntu experimental/20.04_x86_64/ 
Packages [6507 B]
Get:9 http://10.10.21.205:3207/dprepo/ubuntu 20.04_x86_64/ Sources [9066 
B]
Get:10 http://10.10.21.205:3207/dprepo/ubuntu 20.04_x86_64/ Packages 
[23.8 kB]
Get:11 http://security.ubuntu.com/ubuntu focal-security/main amd64 
DEP-11 Metadata [40.6 kB]
Get:12 http://security.ubuntu.com/ubuntu focal-security/universe amd64 
DEP-11 Metadata [66.3 kB]
Get:13 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 
DEP-11 Metadata [2464 B]
Fetched 273 kB in 1s (288 kB/s)
Reading package lists...
Batterie-Status prüfen
Reading package lists...
Building dependency tree...

Semper eget duis at tellus at urna. Enim diam vulputate ut pharetra sit 
amet aliquam id diam. Sollicitudin tempor id eu nisl nunc. Amet justo 
donec enim diam vulputate ut pharetra sit. Massa tempor nec feugiat nisl 
pretium fusce id velit ut. Semper feugiat nibh sed pulvinar proin 
gravida hendrerit lectus. Convallis posuere morbi leo urna molestie at 
elementum eu. Blandit aliquam etiam erat velit scelerisque in. Nisl 
tincidunt eget nullam non nisi est sit amet facilisis. In hendrerit 
gravida rutrum quisque non tellus orci. Et leo duis ut diam quam. Quis 
hendrerit dolor magna eget. Magna eget est lorem ipsum. Sit amet 
volutpat consequat mauris nunc congue nisi vitae. Diam maecenas 
ultricies mi eget. Enim tortor at auctor urna.

Nam at lectus urna duis convallis convallis tellus id interdum. Id 
aliquet lectus proin nibh nisl condimentum id. Vel elit scelerisque 
mauris pellentesque pulvinar pellentesque habitant. Vitae tortor 
condimentum lacinia quis vel eros. Dolor morbi non arcu risus quis 
varius. Nec nam aliquam sem et tortor consequat. Sagittis orci a 
scelerisque purus. Ultricies mi eget mauris pharetra et. Ut aliquam 
purus sit amet luctus venenatis lectus magna. Lectus vestibulum mattis 
ullamcorper velit sed ullamcorper morbi. Eu consequat ac felis donec et. 
Imperdiet dui accumsan sit amet nulla facilisi morbi tempus. Non odio 
euismod lacinia at quis risus. Est ullamcorper eget nulla facilisi etiam 
dignissim diam. Lectus sit amet est placerat. Lectus magna fringilla 
urna porttitor rhoncus dolor purus. Nec feugiat in fermentum posuere 
urna nec tincidunt praesent. Egestas quis ipsum suspendisse ultrices 
gravida dictum fusce. Volutpat sed cras ornare arcu dui vivamus.

Quam viverra orci sagittis eu volutpat odio. Egestas sed tempus urna et 
pharetra pharetra massa massa ultricies. Sagittis nisl rhoncus mattis 
rhoncus urna neque viverra. Gravida dictum fusce ut placerat orci nulla 
pellentesque dignissim. Feugiat nibh sed pulvinar proin gravida 
hendrerit lectus. Molestie ac feugiat sed lectus vestibulum mattis. Sit 
amet purus gravida quis blandit. Massa vitae tortor condimentum lacinia. 
Ultricies lacus sed turpis tincidunt id aliquet. Rhoncus urna neque 
viverra justo nec ultrices. Vulputate odio ut enim blandit volutpat 
maecenas volutpat. Lorem sed risus ultricies tristique.

-- 
My amazing company
--=_e2d4c04609c2d7bd8a014063cc6120fa
Content-Transfer-Encoding: quoted-printable
Content-Type: text/html; charset=UTF-8

<html><head><meta http-equiv=3D"Content-Type" content=3D"text/html; charset=
=3DUTF-8" /></head><body style=3D'font-size: 10pt'>
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod =
tempor incididunt ut labore et dolore magna aliqua. Integer malesuada nunc =
vel risus commodo viverra. In aliquam sem fringilla ut. Enim sed faucibus t=
urpis in eu mi bibendum neque egestas. Quis auctor elit sed vulputate mi si=
t amet mauris. Duis tristique sollicitudin nibh sit amet commodo nulla faci=
lisi nullam. Egestas sed sed risus pretium quam. Maecenas sed enim ut sem=
=2E Id leo in vitae turpis massa sed elementum tempus. Tincidunt lobortis f=
eugiat vivamus at. Commodo odio aenean sed adipiscing. Massa id neque aliqu=
am vestibulum morbi blandit. Mauris cursus mattis molestie a. Morbi non arc=
u risus quis varius quam quisque id. Volutpat consequat mauris nunc congue =
nisi vitae suscipit. Aliquet eget sit amet tellus cras adipiscing enim.</p>
<p>Viverra suspendisse potenti nullam ac tortor vitae. Et pharetra pharetra=
 massa massa. Amet nisl suscipit adipiscing bibendum est ultricies integer =
quis auctor. Vestibulum morbi blandit cursus risus at ultrices. Malesuada n=
unc vel risus commodo viverra maecenas. Dictum at tempor commodo ullamcorpe=
r. Purus gravida quis blandit turpis cursus in hac habitasse platea. Felis =
bibendum ut tristique et egestas. Augue lacus viverra vitae congue eu conse=
quat ac felis. Nunc eget lorem dolor sed viverra ipsum nunc aliquet. Ullamc=
orper velit sed ullamcorper morbi tincidunt ornare massa eget. Est ultricie=
s integer quis auctor elit sed.</p>
<pre><code>apt-get update
Get:1 http://security.ubuntu.com/ubuntu focal-security InRelease [114 kB]
Hit:2 http://de.archive.ubuntu.com/ubuntu focal InRelease
Hit:3 http://de.archive.ubuntu.com/ubuntu focal-updates InRelease
Get:4 http://10.10.21.205:3207/dprepo/ubuntu experimental/20.04_x86_64/ InR=
elease [3820 B]
Hit:5 http://de.archive.ubuntu.com/ubuntu focal-backports InRelease
Get:6 http://10.10.21.205:3207/dprepo/ubuntu 20.04_x86_64/ InRelease [3781 =
B]
Get:7 http://10.10.21.205:3207/dprepo/ubuntu experimental/20.04_x86_64/ Sou=
rces [2710 B]
Get:8 http://10.10.21.205:3207/dprepo/ubuntu experimental/20.04_x86_64/ Pac=
kages [6507 B]
Get:9 http://10.10.21.205:3207/dprepo/ubuntu 20.04_x86_64/ Sources [9066 B]
Get:10 http://10.10.21.205:3207/dprepo/ubuntu 20.04_x86_64/ Packages [23.8 =
kB]
Get:11 http://security.ubuntu.com/ubuntu focal-security/main amd64 DEP-11 M=
etadata [40.6 kB]
Get:12 http://security.ubuntu.com/ubuntu focal-security/universe amd64 DEP-=
11 Metadata [66.3 kB]
Get:13 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 DE=
P-11 Metadata [2464 B]
Fetched 273 kB in 1s (288 kB/s)
Reading package lists...
Batterie-Status pr&uuml;fen
Reading package lists...
Building dependency tree...</code></pre>
<p>Semper eget duis at tellus at urna. Enim diam vulputate ut pharetra sit =
amet aliquam id diam. Sollicitudin tempor id eu nisl nunc. Amet justo donec=
 enim diam vulputate ut pharetra sit. Massa tempor nec feugiat nisl pretium=
 fusce id velit ut. Semper feugiat nibh sed pulvinar proin gravida hendreri=
t lectus. Convallis posuere morbi leo urna molestie at elementum eu. Blandi=
t aliquam etiam erat velit scelerisque in. Nisl tincidunt eget nullam non n=
isi est sit amet facilisis. In hendrerit gravida rutrum quisque non tellus =
orci. Et leo duis ut diam quam. Quis hendrerit dolor magna eget. Magna eget=
 est lorem ipsum. Sit amet volutpat consequat mauris nunc congue nisi vitae=
=2E Diam maecenas ultricies mi eget. Enim tortor at auctor urna.</p>
<p>Nam at lectus urna duis convallis convallis tellus id interdum. Id aliqu=
et lectus proin nibh nisl condimentum id. Vel elit scelerisque mauris pelle=
ntesque pulvinar pellentesque habitant. Vitae tortor condimentum lacinia qu=
is vel eros. Dolor morbi non arcu risus quis varius. Nec nam aliquam sem et=
 tortor consequat. Sagittis orci a scelerisque purus. Ultricies mi eget mau=
ris pharetra et. Ut aliquam purus sit amet luctus venenatis lectus magna. L=
ectus vestibulum mattis ullamcorper velit sed ullamcorper morbi. Eu consequ=
at ac felis donec et. Imperdiet dui accumsan sit amet nulla facilisi morbi =
tempus. Non odio euismod lacinia at quis risus. Est ullamcorper eget nulla =
facilisi etiam dignissim diam. Lectus sit amet est placerat. Lectus magna f=
ringilla urna porttitor rhoncus dolor purus. Nec feugiat in fermentum posue=
re urna nec tincidunt praesent. Egestas quis ipsum suspendisse ultrices gra=
vida dictum fusce. Volutpat sed cras ornare arcu dui vivamus.</p>
<p>Quam viverra orci sagittis eu volutpat odio. Egestas sed tempus urna et =
pharetra pharetra massa massa ultricies. Sagittis nisl rhoncus mattis rhonc=
us urna neque viverra. Gravida dictum fusce ut placerat orci nulla pellente=
sque dignissim. Feugiat nibh sed pulvinar proin gravida hendrerit lectus. M=
olestie ac feugiat sed lectus vestibulum mattis. Sit amet purus gravida qui=
s blandit. Massa vitae tortor condimentum lacinia. Ultricies lacus sed turp=
is tincidunt id aliquet. Rhoncus urna neque viverra justo nec ultrices. Vul=
putate odio ut enim blandit volutpat maecenas volutpat. Lorem sed risus ult=
ricies tristique.</p>
<div id=3D"signature">
<div class=3D"pre" style=3D"margin: 0; padding: 0; font-family: monospace">=
<span class=3D"sig">--&nbsp;<br />My amazing company</span></div>
</div>
</body></html>

--=_e2d4c04609c2d7bd8a014063cc6120fa--
```

### Steps to reproduce the behavior:

* Import above sample and check the original (before import) with the imported article in Zammad

Yes I'm sure this is a bug and no feature request or a general question.

**Comment by uli-heller** [comment]
Hey Marcel,

thanks for creating the issue. How can I provide some coffee to you?

From my point of view, no need to bother about GDPR. There is no personal data
included in the ticket. I grant permission to use any data.

Thx a lot, Uli

**Comment by mgruner** [comment]
Fixed via 278f55a.

**Comment by uli-heller** [comment]
The fix works for me, thanks a lot!
